### PR TITLE
Add optional argument that pass in derived rule when creating a `ADGradientWrapper`

### PR DESCRIPTION
**Pull request by sunxd3**
This would allow us to reuse the already derived rule.

### CODE REVIEW

File: ext/TapirLogDensityProblemsADExt.jl [hunk 1]
```diff
@@ -31,9 +31,11 @@ end
 
 Gradient using algorithmic/automatic differentiation via Tapir.
 """
-function ADgradient(::Val{:Tapir}, ℓ; safety_on::Bool=false)
-    primal_sig = Tuple{typeof(logdensity), typeof(ℓ), Vector{Float64}}
-    rule = Tapir.build_rrule(Tapir.TapirInterpreter(), primal_sig; safety_on)
+function ADgradient(::Val{:Tapir}, ℓ; safety_on::Bool=false, rule=nothing)
```
Comment (sunxd3) [inline ext/TapirLogDensityProblemsADExt.jl:34]: not type stable, what should the type signature be?

Comment (willtebbutt) [reply ext/TapirLogDensityProblemsADExt.jl:34]: What kind of type instabilities are you seeing here?

Comment (sunxd3) [reply ext/TapirLogDensityProblemsADExt.jl:34]: thought need to give some type hint to `rule`, but not actually needed

**Review by willtebbutt — APPROVED** [review]
Thanks for this PR. I'm happy to accept.

Please bump the patch version, and then feel free to merge when CI passes.

**Pull request merged**